### PR TITLE
Update aditof demo to support setFrameType function

### PR DESCRIPTION
**Pull request by SeptimiuVana**

### CODE REVIEW

File: examples/first-frame-network/main.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ int main(int argc, char *argv[]) {
         std::cout << "no frame type available!";
         return 0;
     }
-    status = camera->setFrameType(frameTypes.front());
+    status = camera->setFrameType("depth_only");
```
Comment (dNechita) [inline examples/first-frame-network/main.cpp:76]: I think this change was just to test things, right?

Comment (SeptimiuVana) [reply examples/first-frame-network/main.cpp:76]: If we keep it this way we can use the example with jetson as well, it can be reverted.

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp [hunk 2]
```diff
@@ -188,13 +189,15 @@ aditof::Status CameraFxTof1::initialize() {
 }
 
 aditof::Status CameraFxTof1::start() {
-    // return m_depthSensor->start(); // For now we keep the device open all the time
-    return aditof::Status::OK;
+    return m_depthSensor
+        ->start(); // For now we keep the device open all the time
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:193]: The comments no longer make sense if you uncommented this line.

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp [hunk 3]
```diff
@@ -308,12 +323,18 @@ aditof::Status CameraFxTof1::setFrameType(const std::string &frameType) {
     }
 
     if (m_details.frameType != *frameDetailsIt) {
+        uint16_t afeRegsAddr[2] = {0x4001, 0x7c22};
+        uint16_t afeRegsVal[2] = {0x0006, 0x0004};
+        m_depthSensor->writeAfeRegisters(afeRegsAddr, afeRegsVal, 2);
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:328]: What is happening here? Why is this required?

Comment (SeptimiuVana) [reply sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:328]: We need to turn off the streaming before setting the frame type and then turning it back on.

Comment (dNechita) [reply sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:328]: This needs to be documented next to the code.

File: examples/first-frame-network/main.cpp (hunk 1 above)
Comment (dNechita) [reply examples/first-frame-network/main.cpp:76]: Also, if we keep this, it won't work for remote<->USB<->target setup because in this scenario there is only "depth_ir" (https://github.com/analogdevicesinc/aditof_sdk/blob/master/sdk/src/connections/usb/linux/usb_depth_sensor_linux.cpp#L176 and https://github.com/analogdevicesinc/aditof_sdk/blob/master/sdk/src/connections/usb/windows/usb_depth_sensor_windows.cpp#L523)

Comment (dNechita) [reply examples/first-frame-network/main.cpp:76]: you could set the frame type with the first element from frameTypes. This way it's more generic.

Comment (SeptimiuVana) [reply examples/first-frame-network/main.cpp]: Fixed.

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp (hunk 3 above)
Comment (SeptimiuVana) [reply sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:325]: Fixed.

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp (hunk 2 above)
Comment (SeptimiuVana) [reply sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp]: Fixed.

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp [hunk 4]
```diff
@@ -332,6 +329,7 @@ aditof::Status CameraFxTof1::setFrameType(const std::string &frameType) {
             return status;
         }
         m_details.frameType = *frameDetailsIt;
+        //Turn of the streaming of data
```
Comment (dNechita) [inline sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:332]: off (typo) or should it be on?

File: sdk/src/connections/target/addi9036_sensor.cpp [hunk 5]
```diff
@@ -369,6 +369,16 @@ Addi9036Sensor::setFrameType(const aditof::FrameDetails &details) {
...
+            CLEAR(req);
+	        req.count = 0;
+	        req.type = dev->videoBuffersType;
+	        req.memory = V4L2_MEMORY_MMAP;
+
+	        if (xioctl(dev->fd, VIDIOC_REQBUFS, &req) == -1) {
+	            LOG(WARNING) << "VIDIOC_REQBUFS error "
+	                         << "errno: " << errno << " error: " << strerror(errno);
+	            return Status::GENERIC_ERROR;
+
```
Comment (dNechita) [inline sdk/src/connections/target/addi9036_sensor.cpp]: Why is this needed?

File: sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp (hunk 4 above)
Comment (SeptimiuVana) [reply sdk/src/cameras/ad-fxtof1-ebz/camera_fxtof1.cpp:332]: It should be on. Sorry, i will fix this.

File: sdk/src/connections/target/addi9036_sensor.cpp (hunk 5 above)
Comment (SeptimiuVana) [reply sdk/src/connections/target/addi9036_sensor.cpp]: This was added to clear the buffers from the driver.

Comment (dNechita) [reply sdk/src/connections/target/addi9036_sensor.cpp]: ok. Can you add a short comment with this information?

Comment (SeptimiuVana) [reply sdk/src/connections/target/addi9036_sensor.cpp]: Fixed.

**Review by dNechita — APPROVED** [review]
